### ISSUE TITLE
Why an AMM on the Cosmos Hub?

### ISSUE THREAD

**Issue by zmanian** [cosmosdevs/atom2021#5]
1. I think liquidity between delegation vouchers/staking derivatives is pretty natural use case.

2. For the same reason converting non native fee tokens to ATOMs is another natural use case

**Comment by Hyung-bharvest** [comment]
I personally emphasize future user-bases outside our ecosystem.

Multiple stable pegzones will allow BTC, tokens from Ethereum, BCH, LTC, and many other tokens to participate in our Defi service.

AMM is one of the prime natural usecase for such pegged-in-users from outside Cosmos.
(example : LTC holders can peg in LTC into the hub to swap to USDT and peg out to Ethereum blockchain without any hussle to log in, deposit, trade, withdraw from centralized exchanges with KYC responsibility. I hope this peg-in/swap/pegout should be provided on one-stop interface to provide user convenience competitiveness)

I think our Defi energy should be focused on realization of this "external" interoperability with outside blockchains because it litteraly has much bigger potential user-base in future than internal users inside Cosmos ecosystem.

My vision for the Hub is becoming a crossroad for external blockchains, not becoming a bounded ecosystem with complex derivatives with such small basket. Energy is limited and we should focus on important things, and main effort should be directed to "attract external users".

I think staking derivatives will help activate our bonded capital, but we have more important vision to focus on I think, **the interoperability for island blockchains.** In this context, I think staking derivatives and fee tokens(which will have very small economic significance) are not primary objectives for AMM on the hub.

**Comment by ebuchman** [comment]
> For the same reason converting non native fee tokens to ATOMs is another natural use case

I think this might be necessary sooner than later to simplify mempool design and fee distribution before we have something more advanced to handle the multi token case. Would also create continuous buy pressure for Atom.

This has a big impact on enabling people to use the hub without first acquiring Atoms (eg. if they already have ETH, BTC, etc). so I think it's more critical to the "crossroad for external blockchains" than just "fee tokens(which will have very small economic significance)" might suggest

**Comment by Hyung-bharvest** [comment]
> > For the same reason converting non native fee tokens to ATOMs is another natural use case
> 
> I think this might be necessary sooner than later to simplify mempool design and fee distribution before we have something more advanced to handle the multi token case. Would also create continuous buy pressure for Atom.
> 
> This has a big impact on enabling people to use the hub without first acquiring Atoms (eg. if they already have ETH, BTC, etc). so I think it's more critical to the "crossroad for external blockchains" than just "fee tokens(which will have very small economic significance)" might suggest

1) open option to pay gas with more general tokens like CBTC/CETH/CDAI/CUSDT will be a necessary environment for external users to participate in our Hub ecosystem.

2) Actually this is already possible if validators can quickly adjust the fee option according to such mainstream tokens

3) But I really don't think swapping such fee tokens into atom will make any significant price upward movement.
Simple calculation yields, 100,000 txs/day * $0.01 per tx * 365 day = $365,000 = 0.0365% of total market cap $1B
So, allowing mainstream tokens to pay gas is important, but the price impact on Atom is not.
I think it would be more simple if we can just accept one kind of major stable coin as gas. such as CDAI or CUSDT.
Exchanging CBTC to CDAI or CUSDT will take only 30 seconds if we have basic AMM in our ecosystem.

**Comment by ebuchman** [comment]
Yep the buy pressure is a small point more important thing is to reduce the number of denominations a validator may need to accept gas in by passing everything through an AMM first for either Atom or some stablecoin

**Comment by Hyung-bharvest** [comment]
> Yep the buy pressure is a small point more important thing is to reduce the number of denominations a validator may need to accept gas in by passing everything through an AMM first for either Atom or some stablecoin

I think we will see a service like dust token swap service in binance cex. There will be multiple liquidity providers who are willing to provide swap service from disclosed price list of tokens. More token list he has, more client he receives, more commission and bidask spread earned.

This will be just one of the many expansion of AMM infrastructure.
